tests/unit/conditions/test_address_allowlist_condition.py

```diff
@@ -0,0 +1,135 @@
+import pytest
+from eth_account import Account
+
+from nucypher.policy.conditions.address import AddressAllowlistCondition
+from nucypher.policy.conditions.context import USER_ADDRESS_CONTEXT
+from nucypher.policy.conditions.exceptions import (
+    InvalidCondition,
+    InvalidConditionContext,
+    InvalidConditionLingo,
+)
+from nucypher.policy.conditions.lingo import ConditionType
+
+
+def test_address_allowlist_condition_init():
+    """Test the initialization of AddressAllowlistCondition."""
+    # Create test addresses
+    account1 = Account.create()
+    account2 = Account.create()
+
+    addresses = [account1.address, account2.address]
+
+    # Test successful initialization
+    condition = AddressAllowlistCondition(
+        user_address=USER_ADDRESS_CONTEXT, addresses=addresses
+    )
+    assert condition.condition_type == ConditionType.ADDRESS_ALLOWLIST.value
+    assert set(condition.addresses) == set(addresses)
+
+    # Test with empty addresses list
+    with pytest.raises(InvalidCondition):
+        AddressAllowlistCondition(user_address=USER_ADDRESS_CONTEXT, addresses=[])
+
+    # Test with invalid address
+    with pytest.raises(InvalidCondition):
+        AddressAllowlistCondition(
+            user_address=USER_ADDRESS_CONTEXT,
+            addresses=["not-an-ethereum-address"],
+        )
+
+    # Test with duplicate addresses
+    with pytest.raises(InvalidCondition):
+        AddressAllowlistCondition(
+            user_address=USER_ADDRESS_CONTEXT,
+            addresses=[account1.address, account1.address],
+        )
+
```

Comment:
Could also test a couple other scenarios:
1. invalid `user_address` value (i.e. not the correct context variable value)
2. non-checksummed address in addresses.